sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -310,8 +315,84 @@ public <T> Mono<CosmosItemResponse<T>> createItem(T item, CosmosItemRequestOptio
+            .onErrorResume(throwable -> {
+                Throwable error = throwable instanceof CompletionException ? throwable.getCause() : throwable;
+
+                if (!(error instanceof CosmosException)) {
+
+                    Exception nonCosmosException =
+                        error instanceof Exception ? (Exception) error : new RuntimeException(error);
+                    return Mono.error(nonCosmosException);
+                }
+
+                assert error instanceof CosmosException;
+                CosmosException cosmosException = (CosmosException) error;
+
+                if (cosmosException.getStatusCode() != HttpConstants.StatusCodes.CONFLICT) {
+                    return Mono.error(cosmosException);
+                }
+
+                InternalObjectNode internalObjectNode = InternalObjectNode.fromObjectToInternalObjectNode(item);
+                String itemId = internalObjectNode.getId();
+                RequestOptions requestOptions = ModelBridgeInternal.toRequestOptions(options);
+                @SuppressWarnings("unchecked")
+                Class<T> itemType = (Class<T>) item.getClass();
+
+                Mono<CosmosItemResponse<T>> readMono =
+                    this.getDatabase().getDocClientWrapper()
+                        .readDocument(getItemLink(itemId), requestOptions)
+                        .map(response -> {
+                            CosmosDiagnostics responseDiagnostics = response.getDiagnostics();
+                            if (responseDiagnostics != null &&
+                                responseDiagnostics.getClientSideRequestStatisticsRaw() != null) {
+
+                                CosmosDiagnostics errorDiagnostics = cosmosException.getDiagnostics();
+                                if (errorDiagnostics != null) {
+                                    responseDiagnostics.clientSideRequestStatistics().recordContributingPointOperation(
+                                        errorDiagnostics.getClientSideRequestStatisticsRaw()
+                                    );
+                                }
+
+                                response.addRequestCharge(cosmosException.getRequestCharge());
+                            }
+                            return ModelBridgeInternal
+                                .createCosmosAsyncItemResponse(response, itemType, getItemDeserializer());
+                        })
+                        .single();
+
+                return readMono
+                    .onErrorMap(readThrowable -> cosmosException)
+                    .flatMap(readResponse -> {
+                        if (readResponse.getStatusCode() == 200) {
+                            return Mono.just(itemResponseAccessor.withRemappedStatusCode(
+                                readResponse, 201, cosmosException.getRequestCharge()));
+                        }
+
+                        return Mono.error(cosmosException);
```

Comment:
I guess I still have the question, but it will exists even today, so not a concern, just want to bring up and confirm:

For session consistency, could the reads getting 404/0 if the sessionToken is empt? and in this case, we are just going to throw back 409

Reply:
If the account is stong/bounded the 409 is also strong consistent - at that point we know already that the doc is there
For the subsequent read we own the consistency level - no way to downgrade to eventual/session for that read - it will be a full read using consistency reader for strong/bounded